SPIRE tests failing on Golang 1.14.2

```
--- FAIL: TestValidateSpiffeID (0.00s)
    --- FAIL: TestValidateSpiffeID/test_validate_spiffe_id_invalid_uri (0.00s)
        spiffeid_test.go:258: 
                Error Trace:    spiffeid_test.go:258
                Error:          Error message not equal:
                                expected: "could not parse SPIFFE ID: parse 192.168.2.2:6688: first path segment in URL cannot contain colon"
                                actual  : "could not parse SPIFFE ID: parse \"192.168.2.2:6688\": first path segment in URL cannot contain colon"
                Test:           TestValidateSpiffeID/test_validate_spiffe_id_invalid_uri
FAIL
FAIL    github.com/spiffe/spire/pkg/common/idutil       0.068s
```

This was broken by:
https://github.com/golang/go/commit/64cfe9fe22113cd6bc05a2c5d0cbe872b1b57860
where the error message added quoting.  So SPIRE tests fail on all 1.14 golang releases.

Having tests depend on external error messages is fragile.

--- a/pkg/common/idutil/spiffeid_test.go
+++ b/pkg/common/idutil/spiffeid_test.go
@@ -24,7 +24,7 @@ func TestValidateSpiffeID(t *testing.T) {
 			name:          "test_validate_spiffe_id_invalid_uri",
 			spiffeID:      "192.168.2.2:6688",
 			mode:          AllowAny(),
-			expectedError: "could not parse SPIFFE ID: parse 192.168.2.2:6688: first path segment in URL cannot contain colon",
+			expectedError: "could not parse SPIFFE ID:",
 		},
 		{
 			name:          "test_validate_spiffe_id_invalid_scheme",
@@ -255,7 +255,8 @@ func TestValidateSpiffeID(t *testing.T) {
 			if test.expectedError == "" {
 				assert.NoError(t, err)
 			} else {
-				assert.EqualError(t, err, test.expectedError)
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedError)
 			}
 		})
 	}

